tests/checkers/unittest_unicode/__init__.py

```diff
@@ -0,0 +1,83 @@
+import io
+from pathlib import Path
+
+import pylint.interfaces
+import pylint.testutils
+
+CODEC_AND_MSG = [
+    ("utf-8", tuple()),
+    (
+        "utf-16",
+        (
+            pylint.testutils.MessageTest(
+                msg_id="invalid-unicode-codec",
+                confidence=pylint.interfaces.HIGH,
+                # node=module,
+                line=1,
+                end_line=1,
+                col_offset=None,
+                end_col_offset=None,
+            ),
+        ),
+    ),
+    (
+        "utf-32",
+        (
+            pylint.testutils.MessageTest(
+                msg_id="invalid-unicode-codec",
+                confidence=pylint.interfaces.HIGH,
+                # node=module,
+                line=1,
+                end_line=1,
+                col_offset=None,
+                end_col_offset=None,
+            ),
+        ),
+    ),
+    (
+        "iso-8859-1",
+        (
+            pylint.testutils.MessageTest(
+                msg_id="bad-file-encoding",
+                confidence=pylint.interfaces.HIGH,
+                # node=module,
+                line=1,
+                end_line=1,
+                col_offset=None,
+                end_col_offset=None,
+            ),
+        ),
+    ),
+    (
+        "ascii",
+        (
+            pylint.testutils.MessageTest(
+                msg_id="bad-file-encoding",
+                confidence=pylint.interfaces.HIGH,
+                # node=module,
+                line=1,
+                end_line=1,
+                col_offset=None,
+                end_col_offset=None,
+            ),
+        ),
+    ),
+]
+
+
+class FakeNode:
+    """Just a very simple Faker as astroid crashes in a number of cases we want to lint
+
+    So of course this lints won't work at the moment but still produce a
+    negative result.
+    But as the test can pass, we still don't rely on the behavior, in case it changes
+    in the future.
```

Comment:
```suggestion
    """```suggestion
    """Simple Faker representing a Module node. 
    
    Astroid crashes in a number of cases if we want to lint unsupported encodings.
    So, this is used to test the behaviour of the encoding checker.
    This also makes it so that tests keep passing if Python does support other encodings in the future.
```

Reply:
Reforumlated